src/module/functions/local_function/mod.rs

```diff
@@ -1305,26 +1312,18 @@ fn append_instruction<'context>(
             ctx.alloc_instr(TableCopy { src, dst }, loc);
         }
 
-        Operator::TableInit { segment, table } => {
-            let elem = ctx.indices.get_element(segment).unwrap();
+        Operator::TableInit { elem_index, table } => {
+            let elem = ctx.indices.get_element(elem_index).unwrap();
             let table = ctx.indices.get_table(table).unwrap();
             ctx.alloc_instr(TableInit { elem, table }, loc);
         }
 
-        Operator::ElemDrop { segment } => {
-            let elem = ctx.indices.get_element(segment).unwrap();
+        Operator::ElemDrop { elem_index } => {
+            let elem = ctx.indices.get_element(elem_index).unwrap();
             ctx.alloc_instr(ElemDrop { elem }, loc);
         }
 
-        Operator::ReturnCall { .. }
-        | Operator::ReturnCallIndirect { .. }
-        | Operator::Try { ty: _ }
-        | Operator::Catch { index: _ }
-        | Operator::Throw { index: _ }
-        | Operator::Rethrow { relative_depth: _ }
-        | Operator::Delegate { relative_depth: _ }
-        | Operator::CatchAll => {
-            unimplemented!("not supported")
-        }
+        // TODO: The remaining operators are for some on-going proposals
+        other => unimplemented!("unsupported operator: {:?}", other),
```

Comment:
Rather than a generic `other` could we fully expand this but have all options throwing unsupported operator error? That way future upgrades won't miss additions to this list that may be important to know.